{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build hNET in a neuronal membrane model\n",
    "## Project Background\n",
    "Developing an in silico model of the human Norepinephrine Transporter (hNET) in a neuronal cellular membrane.  Following the process established previously in this Lab to create a model of the human Serotonin Transporter in a neuronal membrane.\n",
    "## NET Homology Model Construction\n",
    "### Introduction\n",
    "To date, there is no structure of human nor-epinephrine transporter (hNET), however there are paralogs in dopamine and serotonin transporters. We will first have to create a homology model of hNET."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Source sequence\n",
    "\n",
    "Server: https://www.uniprot.org/\n",
    "UniProt ID: **P23975**\n",
    "\n",
    "Sequence:\n",
    "3 isoforms have been sequenced.  We will use the canonical sequence **P23975-1** \n",
    "\n",
    ">sp|P23975|SC6A2_HUMAN Sodium-dependent noradrenaline transporter OS=Homo sapiens \n",
    "OX=9606 GN=SLC6A2 PE=1 SV=1 \\\n",
    "MLLARMNPQVQPENNGADTGPEQPLRARKTAELLVVKERNG\\\n",
    "VQCLLAPRDGDAQPRETWGKKIDFLLSVVGFAVDLANVWRF\\\n",
    "PYLCYKNGGGAFLIPYTLFLIIAGMPLFYMELALGQYNREG\\\n",
    "AATVWKICPFFKGVGYAVILIALYVGFYYNVIIAWSLYYLF\\\n",
    "SSFTLNLPWTDCGHTWNSPNCTDPKLLNGSVLGNHTKYSKY\\\n",
    "KFTPAAEFYERGVLHLHESSGIHDIGLPQWQLLLCLMVVVI\\\n",
    "VLYFSLWKGVKTSGKVVWITATLPYFVLFVLLVHGVTLPGA\\\n",
    "SNGINAYLHIDFYRLKEATVWIDAATQIFFSLGAGFGVLIA\\\n",
    "FASYNKFDNNCYRDALLTSSINCITSFVSGFAIFSILGYMA\\\n",
    "HEHKVNIEDVATEGAGLVFILYPEAISTLSGSTFWAVVFFV\\\n",
    "MLLALGLDSSMGGMEAVITGLADDFQVLKRHRKLFTFGVTF\\\n",
    "STFLLALFCITKGGIYVLTLLDTFAAGTSILFAVLMEAIGV\\\n",
    "SWFYGVDRFSNDIQQMMGFRPGLYWRLCWKFVSPAFLLFVV\\\n",
    "VVSIINFKPLTYDDYIFPPWANWVGWGIALSSMVLVPIYVI\\\n",
    "YKFLSTQGSLWERLAYGITPENEHHLVAQRDIRQFQLQHWL\\\n",
    "AI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Homology Modelling\n",
    "\n",
    "Searching for known structures for proteins containing similar amino acid sequences to those of our target protein (hNET) using the Phyre2 service.      \n",
    "\n",
    "Server: http://www.sbg.bio.ic.ac.uk/phyre2\n",
    "\n",
    "PDB homolog structure: **PHYRE2 – Intensive mode**\n",
    "PDB Selected templates:  \n",
    "1: **c4m48A_**  \n",
    "PDB header:transport protein  \n",
    "Chain: A: PDB Molecule:transporter;  \n",
    "PDBTitle: x-ray structure of dopamine transporter elucidates antidepressant2 mechanism  \n",
    "2: **c5i6xA_**  \n",
    "PDB header:membrane protein  \n",
    "Chain: A: PDB Molecule:sodium-dependent serotonin transporter;  \n",
    "PDBTitle: x-ray structure of the ts3 human serotonin transporter complexed with2 paroxetine at the central site  \n",
    "\n",
    "**Result**: 88% of residues modelled at >90% confidence from two templates with the following 75 residues modelled ab initio  \n",
    "1-50:  \n",
    "202-207:  \n",
    "597-617:  \n",
    "\n",
    "NB:\n",
    "The 50 residue N-terminus loop not resolved from homology was removed (using a text editor) but not capped.\n",
    "\n",
    "The 20 residue C-terminus loop not resolved from homology was removed (using a text editor) but not capped.\n",
    "\n",
    "RESULT: **hNET_truncated.pdb**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a file to automate rendering an image of the protein, \n",
    "# display projection Orthographic\n",
    "# setting \"bead\" style, \n",
    "# setting the background to white,\n",
    "# turning off fthe XYZ axes,  \n",
    "# and taking a snapshot \n",
    "!printf \"display projection Orthographic\\nmol modcolor 0 0 Structure\\nmol modstyle 0 0 NewRibbons 1.000000 12.000000 3.000000\\naxes location Off\\ncolor Display Background white\\nrender snapshot hNET_truncated.tga {convert hNET_truncated.tga jpg:hNET_truncated.jpg; rm hNET_truncated.tga}\\nexit\" > output.tcl\n",
    "# run VMD to render the image \n",
    "output = !vmd hNET_truncated.pdb -e output.tcl \n",
    "\n",
    "# clean up temporary files\n",
    "!rm output.tcl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Rendered protein\n",
    "<img src=\"hNET_truncated.jpg\" alt=\"Rendered protein\" width=\"400\"/>\n",
    "NB: Protein is not necessarily oriented"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The following non-deterministic can be repeated for replicants.\n",
    "\n",
    "Starting files required:\n",
    "* folder martini.ff\n",
    "  * martini_v2.2.itp\n",
    "  * martini_v2.2P.itp\n",
    "  * martini_v2.2refP.itp\n",
    "  * martini_v2.0_brain_complex_b5-GMs_old.itp\n",
    "* ../martinize.py \n",
    "* ../insane.py \n",
    "* **hNET_truncated.pdb** "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Convert atomistic protein structure to coarse grained model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO       MARTINIZE, script version 2.6_3\n",
      "INFO       If you use this script please cite:\n",
      "INFO       de Jong et al., J. Chem. Theory Comput., 2013, DOI:10.1021/ct300646g\n",
      "INFO       Chain termini will be charged\n",
      "INFO       Residues at chain brakes will not be charged\n",
      "INFO       The martini22 forcefield will be used.\n",
      "INFO       Local elastic bonds will be used for extended regions.\n",
      "INFO       Position restraints will be generated.\n",
      "WARNING    Position restraints are only enabled if -DPOSRES is set in the MDP file\n",
      "INFO       Read input structure from file.\n",
      "INFO       Input structure is a PDB file.\n",
      "INFO       Found 1 chains:\n",
      "INFO          1:    (), 4340 atoms in 546 residues.\n",
      "INFO       Total size of the system: 546 residues.\n",
      "INFO       Writing coarse grained structure.\n",
      "INFO       Writing index file.\n",
      "INFO       Writing trajectory index file.\n",
      "INFO       (Average) Secondary structure has been determined (see head of .itp-file).\n",
      "INFO       Created coarsegrained topology\n",
      "INFO       Written 1 ITP file\n",
      "INFO       Output contains 1 molecules:\n",
      "INFO          1->  hNET_CG (chain  )\n",
      "INFO       Written topology files\n",
      "INFO       Note: Cysteine bonds are 0.24 nm constraints, instead of the published 0.39nm/5000kJ/mol.\n",
      "\n",
      "\tThere you are. One MARTINI. Shaken, not stirred.\n",
      "\n",
      "\n",
      " The three-martini lunch is the epitome of American efficiency. Where else can you get an earful, a bellyful and a snootful at the same time? \n",
      "                                                                --Gerald R. Ford \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Use martinize script to make coarse grained model of the protein\n",
    "# INPUT: hNET_truncated.pdb atomistic structure file\n",
    "# OUTPUT: hNET_CG.pdb coarse grained structure file\n",
    "# OUTPUT: hNET_CG.top topology file including references to topology include (ITP) files\n",
    "# OUTPUT: hNET_CG.itp topology include file for the coarse grained representation of this protein\n",
    "# OUTPUT: \n",
    "# OUTPUT: hNET_CG_mapping.ndx per bead mapping \n",
    "# OPTION: -dssp mkdssp is used to generate secondary structure \n",
    "# OPTION: -ff Martini 2.2 forcefield is used\n",
    "# OPTION: -p position restraints Backbone\n",
    "# OPTION: -elastic bonds are used\n",
    "!python3 ../martinize.py -f hNET_truncated.pdb -o hNET_CG.top -x hNET_CG.pdb -n hNET_CG.ndx -nmap hNET_CG_mapping.ndx -dssp mkdssp -name hNET_CG -ff martini22 -p backbone -elastic\n",
    "\n",
    "# create a file to automate rendering an image of the protein, \n",
    "# display projection Orthographic\n",
    "# setting \"bead\" style, \n",
    "# setting the background to white,\n",
    "# turning off the XYZ axes,  \n",
    "# and taking a snapshot \n",
    "!printf \"display projection Orthographic\\nmol modstyle 0 0 Beads 1.000000 12.000000\\naxes location Off\\ncolor Display Background white\\nrender snapshot hNET_CG.tga {convert hNET_CG.tga jpg:hNET_CG.jpg; rm hNET_CG.tga}\\nexit\" > output.tcl\n",
    "# run VMD to render the image \n",
    "output = !vmd hNET_CG.pdb -e output.tcl \n",
    "\n",
    "# clean up temporary files\n",
    "!rm output.tcl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Rendered protein using Coarse grained beads\n",
    "<img src=\"./hNET_CG.jpg\" alt=\"Rendered protein using Coarse grained beads\" width=\"400\"/>\n",
    "NB: Protein is not necessarily oriented"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Insert protein into Neuronal membrane"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "; X: 16.600 (21 lipids) Y: 19.400 (25 lipids)\n",
      "; 486 lipids in upper leaflet, 489 lipids in lower leaflet\n",
      "; NDX Solute 1 1263\n",
      "; Charge of protein: -1.000000\n",
      "; NDX Membrane 1264 11036\n",
      "; Charge of membrane: -12.000000\n",
      "; Total charge: -13.000000\n",
      "; NDX Solvent 11037 110921\n",
      "; NDX System 1 110921\n"
     ]
    }
   ],
   "source": [
    "# Insert protein into Neuronal membrane\n",
    "!python2 ../insane.py -f hNET_CG.pdb -o hNET_neuronal.gro -p hNET_neuronal.top -d 0 -x 16.6 -y 19.4 -z 16 -sol PW -center -charge 0 -u CHOL:0.444 -u POPC:0.087 -u DPSM:0.058 -u DPPC:0.053 -u PUPE:0.05 -u DPGS:0.049 -u PAPC:0.046 -u PAPE:0.031 -u DOPC:0.022 -u PUPC:0.017 -u POPE:0.013 -u PNSM:0.013 -u PBSM:0.011 -u PNGS:0.011 -u DPG1:0.009 -u DPG3:0.009 -u DBGS:0.009 -u OAPE:0.007 -u OUPE:0.007 -u POSM:0.007 -u PFPC:0.006 -u OIPC:0.006 -u POGS:0.006 -u OUPC:0.004 -u DPCE:0.004 -u PADG:0.003 -u DBG1:0.002 -u PNG1:0.002 -u DBG3:0.002 -u PNG3:0.002 -u PPC:0.002 -u OIPE:0.001 -u POG1:0.001 -u POG3:0.001 -u DBCE:0.001 -u PNCE:0.001 -u IPC:0.001 -u PPE:0.001 -u IPE:0.001 -u PODG:0.001 -l DPPC:0.03 -l CHOL:0.446 -l PUPE:0.097 -l PAPE:0.061 -l POPC:0.049 -l PUPS:0.034 -l PAPS:0.028 -l PAPC:0.026 -l POPE:0.025 -l POPS:0.025 -l PUPI:0.02 -l DPSM:0.015 -l OUPE:0.014 -l OAPE:0.013 -l POPI:0.013 -l PAPI:0.013 -l DOPC:0.012 -l PUPC:0.01 -l OUPS:0.007 -l DPPS:0.005 -l PIPI:0.005 -l DPCE:0.004 -l PFPC:0.003 -l OIPC:0.003 -l OIPE:0.003 -l PNSM:0.003 -l PBSM:0.003 -l PAPA:0.003 -l PAP1:0.003 -l PAP2:0.003 -l PAP3:0.003 -l PADG:0.003 -l OUPC:0.002 -l POSM:0.002 -l POP1:0.002 -l POP2:0.002 -l POP3:0.002 -l IPE:0.002 -l POPA:0.001 -l DBCE:0.001 -l PNCE:0.001 -l PPC:0.001 -l IPC:0.001 -l PPE:0.001 -l PODG:0.001 -orient\n",
    "\n",
    "# use GMX to convert GRO file into a PDB\n",
    "output = !gmx editconf -f hNET_neuronal.gro -o  hNET_neuronal.pdb\n",
    "\n",
    "# create a file to automate rendering an image of the protein, \n",
    "# hide water molecules\n",
    "# rotate display to view protein \n",
    "# display projection Orthographic\n",
    "# setting \"bead\" style, \n",
    "# setting the background to white,\n",
    "# turning of fthe XYZ axes,  \n",
    "# and taking a snapshot \n",
    "!printf \"mol modselect 0 0 not resname PW\\nrotate x by 120\\nrotate y by 45\\ndisplay projection Orthographic\\nmol modstyle 0 0 Beads 1.000000 12.000000\\naxes location Off\\ncolor Display Background white\\nrender snapshot hNET_neuronal.tga {convert hNET_neuronal.tga jpg:hNET_neuronal.jpg; rm hNET_neuronal.tga}\\nexit\" > output.tcl\n",
    "# run VMD to render the image \n",
    "output = !vmd hNET_neuronal.pdb -e output.tcl \n",
    "\n",
    "# clean up temporary files\n",
    "!rm output.tcl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Rendered protein using Coarse grained beads\n",
    "<img src=\"./hNET_neuronal.jpg\" alt=\"Rendered protein in a neuronal membrane\" width=\"400\"/>\n",
    "NB: Protein should now be oriented "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare for compilation\n",
    "by adding the include file for the protein to the topology file and changing the generic molecule named protein to the specific element we are using (nHET_CG)\n",
    "\n",
    "NB: Structure (pdb) file generated by martinize has different (older) bead names than used in the martini 2.2 itp file so these must be updated manually\n",
    "\n",
    "Replace molecule in hNET_CG.itp (generated by martinize.py) named AC1 with C1\n",
    "Replace molecule in hNET_CG.itp (generated by martinize.py) named AC2 with C2 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Replace molecule in hNET_CG.itp (generated by martinize.py) named AC1 with C1\n",
    "!sed -i 's/AC1/C1/g' hNET_CG.itp \n",
    "\n",
    "#Replace molecule in hNET_CG.itp (generated by martinize.py) named AC2 with C2\n",
    "!sed -i 's/AC2/C2/g' hNET_CG.itp \n",
    "\n",
    "#Move hNET_CG.itp (generated by martinize.py) to martini.ff folder\n",
    "!cp hNET_CG.itp martini.ff/.\n",
    "\n",
    "#Replace molceule named “Protein” in hNET_neuronal.top (generated by insane.py) with “hNET_CG” (or the molcules name in hNET_CG.itp)  \n",
    "!sed -i 's/Protein/hNET_CG/' hNET_neuronal.top \n",
    "\n",
    "# Add hNET_CG.itp (generated by martinize.py) to  hNET_neuronal.top (generated by insane.py)  \n",
    "!sed -i '3 a #include \"martini.ff/hNET_CG.itp\"'  hNET_neuronal.top "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compile the system\n",
    "\n",
    "by adding the include file for the protein to the topology file and changing the generic molecule named protein to the specific element we are using (nHET_CG)\n",
    "\n",
    "NB: Structure (pdb) file generated by martinize has different (older) bead names than used in the martini 2.2 itp file so these must be updated manually\n",
    "\n",
    "Replace molecule in hNET_CG.itp (generated by martinize.py) named AC1 with C1  \n",
    "Replace molecule in hNET_CG.itp (generated by martinize.py) named AC2 with C2  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      :-) GROMACS - gmx grompp, 2018.1 (-:\n",
      "\n",
      "                            GROMACS is written by:\n",
      "     Emile Apol      Rossen Apostolov      Paul Bauer     Herman J.C. Berendsen\n",
      "    Par Bjelkmar    Aldert van Buuren   Rudi van Drunen     Anton Feenstra  \n",
      "  Gerrit Groenhof    Aleksei Iupinov   Christoph Junghans   Anca Hamuraru   \n",
      " Vincent Hindriksen Dimitrios Karkoulis    Peter Kasson        Jiri Kraus    \n",
      "  Carsten Kutzner      Per Larsson      Justin A. Lemkul    Viveca Lindahl  \n",
      "  Magnus Lundborg   Pieter Meulenhoff    Erik Marklund      Teemu Murtola   \n",
      "    Szilard Pall       Sander Pronk      Roland Schulz     Alexey Shvetsov  \n",
      "   Michael Shirts     Alfons Sijbers     Peter Tieleman    Teemu Virolainen \n",
      " Christian Wennberg    Maarten Wolf   \n",
      "                           and the project leaders:\n",
      "        Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel\n",
      "\n",
      "Copyright (c) 1991-2000, University of Groningen, The Netherlands.\n",
      "Copyright (c) 2001-2017, The GROMACS development team at\n",
      "Uppsala University, Stockholm University and\n",
      "the Royal Institute of Technology, Sweden.\n",
      "check out http://www.gromacs.org for more information.\n",
      "\n",
      "GROMACS is free software; you can redistribute it and/or modify it\n",
      "under the terms of the GNU Lesser General Public License\n",
      "as published by the Free Software Foundation; either version 2.1\n",
      "of the License, or (at your option) any later version.\n",
      "\n",
      "GROMACS:      gmx grompp, version 2018.1\n",
      "Executable:   /usr/bin/gmx\n",
      "Data prefix:  /usr\n",
      "Working dir:  /home/richard/projects/hNET_neuronal/report\n",
      "Command line:\n",
      "  gmx grompp -f Minimization1.mdp -c hNET_neuronal.gro -p hNET_neuronal.top -o hNET_neuronal_EM.tpr -r hNET_neuronal.gro\n",
      "\n",
      "Ignoring obsolete mdp entry 'title'\n",
      "Setting the LD random seed to 206651080\n",
      "Generated 1 of the 741 non-bonded parameter combinations\n",
      "Excluding 1 bonded neighbours molecule type 'hNET_CG'\n",
      "Excluding 1 bonded neighbours molecule type 'CHOL'\n",
      "Excluding 1 bonded neighbours molecule type 'POPC'\n",
      "Excluding 1 bonded neighbours molecule type 'DPSM'\n",
      "Excluding 1 bonded neighbours molecule type 'DPPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPE'\n",
      "Excluding 1 bonded neighbours molecule type 'DPGS'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPE'\n",
      "Excluding 1 bonded neighbours molecule type 'DOPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPC'\n",
      "Excluding 1 bonded neighbours molecule type 'POPE'\n",
      "Excluding 1 bonded neighbours molecule type 'PNSM'\n",
      "Excluding 1 bonded neighbours molecule type 'PBSM'\n",
      "Excluding 1 bonded neighbours molecule type 'PNGS'\n",
      "Excluding 1 bonded neighbours molecule type 'DPG1'\n",
      "Excluding 1 bonded neighbours molecule type 'DPG3'\n",
      "Excluding 1 bonded neighbours molecule type 'DBGS'\n",
      "Excluding 1 bonded neighbours molecule type 'OAPE'\n",
      "Excluding 1 bonded neighbours molecule type 'OUPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POSM'\n",
      "Excluding 1 bonded neighbours molecule type 'PFPC'\n",
      "Excluding 1 bonded neighbours molecule type 'OIPC'\n",
      "Excluding 1 bonded neighbours molecule type 'POGS'\n",
      "Excluding 1 bonded neighbours molecule type 'OUPC'\n",
      "Excluding 1 bonded neighbours molecule type 'DPCE'\n",
      "Excluding 1 bonded neighbours molecule type 'PADG'\n",
      "Excluding 1 bonded neighbours molecule type 'DBG1'\n",
      "Excluding 1 bonded neighbours molecule type 'PNG1'\n",
      "Excluding 1 bonded neighbours molecule type 'DBG3'\n",
      "Excluding 1 bonded neighbours molecule type 'PNG3'\n",
      "Excluding 1 bonded neighbours molecule type 'PPC'\n",
      "Excluding 1 bonded neighbours molecule type 'OIPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POG1'\n",
      "Excluding 1 bonded neighbours molecule type 'POG3'\n",
      "Excluding 1 bonded neighbours molecule type 'DBCE'\n",
      "Excluding 1 bonded neighbours molecule type 'PNCE'\n",
      "Excluding 1 bonded neighbours molecule type 'IPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PPE'\n",
      "Excluding 1 bonded neighbours molecule type 'IPE'\n",
      "Excluding 1 bonded neighbours molecule type 'PODG'\n",
      "Excluding 1 bonded neighbours molecule type 'DPPC'\n",
      "Excluding 1 bonded neighbours molecule type 'CHOL'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPE'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPS'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPS'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPC'\n",
      "Excluding 1 bonded neighbours molecule type 'POPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POPS'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPI'\n",
      "Excluding 1 bonded neighbours molecule type 'DPSM'\n",
      "Excluding 1 bonded neighbours molecule type 'OUPE'\n",
      "Excluding 1 bonded neighbours molecule type 'OAPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POPI'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPI'\n",
      "Excluding 1 bonded neighbours molecule type 'DOPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PUPC'\n",
      "Excluding 1 bonded neighbours molecule type 'OUPS'\n",
      "Excluding 1 bonded neighbours molecule type 'DPPS'\n",
      "Excluding 1 bonded neighbours molecule type 'PIPI'\n",
      "Excluding 1 bonded neighbours molecule type 'DPCE'\n",
      "Excluding 1 bonded neighbours molecule type 'PFPC'\n",
      "Excluding 1 bonded neighbours molecule type 'OIPC'\n",
      "Excluding 1 bonded neighbours molecule type 'OIPE'\n",
      "Excluding 1 bonded neighbours molecule type 'PNSM'\n",
      "Excluding 1 bonded neighbours molecule type 'PBSM'\n",
      "Excluding 1 bonded neighbours molecule type 'PAPA'\n",
      "Excluding 1 bonded neighbours molecule type 'PAP1'\n",
      "Excluding 1 bonded neighbours molecule type 'PAP2'\n",
      "Excluding 1 bonded neighbours molecule type 'PAP3'\n",
      "Excluding 1 bonded neighbours molecule type 'PADG'\n",
      "Excluding 1 bonded neighbours molecule type 'OUPC'\n",
      "Excluding 1 bonded neighbours molecule type 'POSM'\n",
      "Excluding 1 bonded neighbours molecule type 'POP1'\n",
      "Excluding 1 bonded neighbours molecule type 'POP2'\n",
      "Excluding 1 bonded neighbours molecule type 'POP3'\n",
      "Excluding 1 bonded neighbours molecule type 'IPE'\n",
      "Excluding 1 bonded neighbours molecule type 'POPA'\n",
      "Excluding 1 bonded neighbours molecule type 'DBCE'\n",
      "Excluding 1 bonded neighbours molecule type 'PNCE'\n",
      "Excluding 1 bonded neighbours molecule type 'PPC'\n",
      "Excluding 1 bonded neighbours molecule type 'IPC'\n",
      "Excluding 1 bonded neighbours molecule type 'PPE'\n",
      "Excluding 1 bonded neighbours molecule type 'PODG'\n",
      "Excluding 1 bonded neighbours molecule type 'PW'\n",
      "\n",
      "NOTE 1 [file hNET_neuronal.top, line 98]:\n",
      "  System has non-zero total charge: -95.000000\n",
      "  Total charge should normally be an integer. See\n",
      "  http://www.gromacs.org/Documentation/Floating_Point_Arithmetic\n",
      "  for discussion on how close it should be to an integer.\n",
      "  \n",
      "\n",
      "\n",
      "Cleaning up constraints and constant bonded interactions with virtual sites\n",
      "Removing all charge groups because cutoff-scheme=Verlet\n",
      "Analysing residue names:\n",
      "There are:   546    Protein residues\n",
      "There are: 34221      Other residues\n",
      "Analysing Protein...\n",
      "Analysing residues not classified as Protein/DNA/RNA/Water and splitting into groups...\n",
      "Number of degrees of freedom in T-Coupling group System is 259033.00\n",
      "This run will generate roughly 14 Mb of data\n",
      "\n",
      "There was 1 note\n",
      "\n",
      "Back Off! I just backed up hNET_neuronal_EM.tpr to ./#hNET_neuronal_EM.tpr.6#\n",
      "\n",
      "GROMACS reminds you: \"The only greatness for man is immortality.\" (James Dean)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!gmx grompp -f Minimization1.mdp -c hNET_neuronal.gro -p hNET_neuronal.top -o hNET_neuronal_EM.tpr -r  hNET_neuronal.gro"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Minimize the system "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      :-) GROMACS - gmx mdrun, 2018.1 (-:\n",
      "\n",
      "                            GROMACS is written by:\n",
      "     Emile Apol      Rossen Apostolov      Paul Bauer     Herman J.C. Berendsen\n",
      "    Par Bjelkmar    Aldert van Buuren   Rudi van Drunen     Anton Feenstra  \n",
      "  Gerrit Groenhof    Aleksei Iupinov   Christoph Junghans   Anca Hamuraru   \n",
      " Vincent Hindriksen Dimitrios Karkoulis    Peter Kasson        Jiri Kraus    \n",
      "  Carsten Kutzner      Per Larsson      Justin A. Lemkul    Viveca Lindahl  \n",
      "  Magnus Lundborg   Pieter Meulenhoff    Erik Marklund      Teemu Murtola   \n",
      "    Szilard Pall       Sander Pronk      Roland Schulz     Alexey Shvetsov  \n",
      "   Michael Shirts     Alfons Sijbers     Peter Tieleman    Teemu Virolainen \n",
      " Christian Wennberg    Maarten Wolf   \n",
      "                           and the project leaders:\n",
      "        Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel\n",
      "\n",
      "Copyright (c) 1991-2000, University of Groningen, The Netherlands.\n",
      "Copyright (c) 2001-2017, The GROMACS development team at\n",
      "Uppsala University, Stockholm University and\n",
      "the Royal Institute of Technology, Sweden.\n",
      "check out http://www.gromacs.org for more information.\n",
      "\n",
      "GROMACS is free software; you can redistribute it and/or modify it\n",
      "under the terms of the GNU Lesser General Public License\n",
      "as published by the Free Software Foundation; either version 2.1\n",
      "of the License, or (at your option) any later version.\n",
      "\n",
      "GROMACS:      gmx mdrun, version 2018.1\n",
      "Executable:   /usr/bin/gmx\n",
      "Data prefix:  /usr\n",
      "Working dir:  /home/richard/projects/hNET_neuronal/report\n",
      "Command line:\n",
      "  gmx mdrun -v -deffnm hNET_neuronal_EM\n",
      "\n",
      "\n",
      "Back Off! I just backed up hNET_neuronal_EM.log to ./#hNET_neuronal_EM.log.9#\n",
      "Compiled SIMD: SSE2, but for this host/run AVX2_256 might be better (see log).\n",
      "The current CPU can measure timings more accurately than the code in\n",
      "gmx mdrun was configured to use. This might affect your simulation\n",
      "speed as accurate timings are needed for load-balancing.\n",
      "Please consider rebuilding gmx mdrun with the GMX_USE_RDTSCP=ON CMake option.\n",
      "Reading file hNET_neuronal_EM.tpr, VERSION 2018.1 (single precision)\n",
      "\n",
      "Using 1 MPI thread\n",
      "Using 8 OpenMP threads \n",
      "\n",
      "\n",
      "Step -1, time -0.01 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 2.365362, max 360.684418 (between atoms 11008 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "\n",
      "Back Off! I just backed up step-1b.pdb to ./#step-1b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step-1c.pdb to ./#step-1c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "\n",
      "Back Off! I just backed up hNET_neuronal_EM.trr to ./#hNET_neuronal_EM.trr.9#\n",
      "\n",
      "Back Off! I just backed up hNET_neuronal_EM.edr to ./#hNET_neuronal_EM.edr.9#\n",
      "\n",
      "Steepest Descents:\n",
      "   Tolerance (Fmax)   =  1.00000e+01\n",
      "   Number of steps    =        10000\n",
      "Step=    0, Dmax= 1.0e-02 nm, Epot=  1.13358e+09 Fmax= 2.89610e+10, atom= 5756\n",
      "\n",
      "Step 1, time 0.01 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.479004 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step1b.pdb to ./#step1b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step1c.pdb to ./#step1c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    1, Dmax= 1.0e-02 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 2, time 0.02 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478992 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step2b.pdb to ./#step2b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step2c.pdb to ./#step2c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    2, Dmax= 5.0e-03 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 3, time 0.03 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step3b.pdb to ./#step3b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step3c.pdb to ./#step3c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    3, Dmax= 2.5e-03 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 4, time 0.04 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step4b.pdb to ./#step4b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step4c.pdb to ./#step4c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    4, Dmax= 1.2e-03 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 5, time 0.05 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478985 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step5b.pdb to ./#step5b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step5c.pdb to ./#step5c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    5, Dmax= 6.2e-04 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 6, time 0.06 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step6b.pdb to ./#step6b.pdb.9#\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Back Off! I just backed up step6c.pdb to ./#step6c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    6, Dmax= 3.1e-04 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 7, time 0.07 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step7b.pdb to ./#step7b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step7c.pdb to ./#step7c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    7, Dmax= 1.6e-04 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 8, time 0.08 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step8b.pdb to ./#step8b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step8c.pdb to ./#step8c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    8, Dmax= 7.8e-05 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 9, time 0.09 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step9b.pdb to ./#step9b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step9c.pdb to ./#step9c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=    9, Dmax= 3.9e-05 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 10, time 0.1 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step10b.pdb to ./#step10b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step10c.pdb to ./#step10c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=   10, Dmax= 2.0e-05 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 11, time 0.11 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step11b.pdb to ./#step11b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step11c.pdb to ./#step11c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=   11, Dmax= 9.8e-06 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 12, time 0.12 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step12b.pdb to ./#step12b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step12c.pdb to ./#step12c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=   12, Dmax= 4.9e-06 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 13, time 0.13 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step13b.pdb to ./#step13b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step13c.pdb to ./#step13c.pdb.9#\n",
      "Wrote pdb files with previous and current coordinates\n",
      "Step=   13, Dmax= 2.4e-06 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Step 14, time 0.14 (ps)  LINCS WARNING\n",
      "relative constraint deviation after LINCS:\n",
      "rms 0.383720, max 43.478989 (between atoms 11009 and 11010)\n",
      "bonds that rotated more than 90 degrees:\n",
      " atom 1 atom 2  angle  previous, current, constraint length\n",
      "  10976  10979  138.3    4.6858   8.2450      0.3000\n",
      "  10991  10995  160.6    7.8731   8.4476      0.4000\n",
      "  10991  10993  179.3    6.5751   8.0874      0.4000\n",
      "  10992  10993  171.5    3.9885   5.8189      0.4000\n",
      "  10993  10996  127.2    4.4299   4.2630      0.3100\n",
      "  11008  11010  127.7    9.5068  13.7725      0.4000\n",
      "  11008  11014  142.2    3.5855  10.7202      0.4000\n",
      "  11009  11010  110.9   10.3531  17.7916      0.4000\n",
      "  11009  11013  174.7   10.9163   3.8982      0.3500\n",
      "  11010  11013  142.3    7.5327   6.0141      0.3100\n",
      "\n",
      "Back Off! I just backed up step14b.pdb to ./#step14b.pdb.9#\n",
      "\n",
      "Back Off! I just backed up step14c.pdb to ./#step14c.pdb.9#\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote pdb files with previous and current coordinates\n",
      "Step=   14, Dmax= 1.2e-06 nm, Epot=          inf Fmax= 0.00000e+00, atom= 1\n",
      "Energy minimization has stopped, but the forces have not converged to the\n",
      "requested precision Fmax < 10 (which may not be possible for your system). It\n",
      "stopped because the algorithm tried to make a new step whose size was too\n",
      "small, or there was no change in the energy since last step. Either way, we\n",
      "regard the minimization as converged to within the available machine\n",
      "precision, given your starting configuration and EM parameters.\n",
      "\n",
      "Double precision normally gives you higher accuracy, but this is often not\n",
      "needed for preparing to run molecular dynamics.\n",
      "You might need to increase your constraint accuracy, or turn\n",
      "off constraints altogether (set constraints = none in mdp file)\n",
      "\n",
      "writing lowest energy coordinates.\n",
      "\n",
      "Back Off! I just backed up hNET_neuronal_EM.gro to ./#hNET_neuronal_EM.gro.9#\n",
      "\n",
      "Steepest Descents converged to machine precision in 15 steps,\n",
      "but did not reach the requested Fmax < 10.\n",
      "Potential Energy  =  1.1335799e+09\n",
      "Maximum force     =  2.8961004e+10 on atom 5756\n",
      "Norm of force     =  1.3500975e+08\n",
      "\n",
      "GROMACS reminds you: \"Pump Up the Volume Along With the Tempo\" (Jazzy Jeff)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!export GMX_MAXCONSTRWARN=-1\n",
    "!export GMX_SUPPRESS_DUMP=1\n",
    "!gmx mdrun -v -deffnm hNET_neuronal_EM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
